TLM/TLM/Manager/Impl/JunctionRestrictionsManager.cs

```diff
@@ -13,492 +13,481 @@
 #endif
-                return res;
-            }
+				return res;
+			}
 
-            bool ret = (node.m_flags & (NetNode.Flags.End | NetNode.Flags.OneWayOut)) != NetNode.Flags.None;
+			bool ret = (node.m_flags & (NetNode.Flags.End | NetNode.Flags.OneWayOut)) != NetNode.Flags.None;
 
-            if (!ret && Options.allowUTurns) {
-                ret = (node.m_flags & (NetNode.Flags.Junction | NetNode.Flags.Transition)) != NetNode.Flags.None;
-            }
+			if (!ret && Options.allowUTurns) {
+				ret = (node.m_flags & (NetNode.Flags.Junction | NetNode.Flags.Transition)) != NetNode.Flags.None;
+			}
 
 #if DEBUG
-            if (debug)
-                Log._Debug($"JunctionRestrictionsManager.GetDefaultUturnAllowed({segmentId}, {startNode}): Setting is configurable. ret={ret}, flags={node.m_flags}");
+			if (debug)
+				Log._Debug($"JunctionRestrictionsManager.GetDefaultUturnAllowed({segmentId}, {startNode}): Setting is configurable. ret={ret}, flags={node.m_flags}");
 #endif
 
-            return ret;
-        }
+			return ret;
+		}
 
-        public bool IsUturnAllowed(ushort segmentId, bool startNode) {
-            return SegmentFlags[segmentId].IsUturnAllowed(startNode);
-        }
+		public bool IsUturnAllowed(ushort segmentId, bool startNode) {
+			return SegmentFlags[segmentId].IsUturnAllowed(startNode);
+		}
 
-        public bool IsTurnOnRedAllowedConfigurable(ushort segmentId, bool startNode, ref NetNode node) {
+		public bool IsTurnOnRedAllowedConfigurable(ushort segmentId, bool startNode, ref NetNode node) {
 #if DEBUG
-            bool debug = GlobalConfig.Instance.Debug.Switches[11];
+			bool debug = GlobalConfig.Instance.Debug.Switches[11];
 #endif
-	        if (!Options.turnOnRed)
-		        return false;
-	        
-            SegmentGeometry segGeo = SegmentGeometry.Get(segmentId);
-            SegmentEndGeometry endGeo = segGeo?.GetEnd(startNode);
-
-            bool ret = (node.m_flags & (NetNode.Flags.TrafficLights)) != NetNode.Flags.None &&
-                    node.Info?.m_class?.m_service != ItemClass.Service.Beautification &&
-                    !LogicUtil.CheckFlags((uint)node.m_flags, (uint)(NetNode.Flags.LevelCrossing)) &&
-                    !endGeo.OutgoingOneWay &&
-                    segGeo?.HasValidTurnOnRedOutgoingSegment(endGeo) == true;
+
+			SegmentGeometry segGeo = SegmentGeometry.Get(segmentId);
+			SegmentEndGeometry endGeo = segGeo?.GetEnd(startNode);
+
+			bool ret = (node.m_flags & (NetNode.Flags.TrafficLights | NetNode.Flags.LevelCrossing)) == NetNode.Flags.TrafficLights &&
+					node.Info?.m_class?.m_service != ItemClass.Service.Beautification &&
+					!endGeo.OutgoingOneWay &&
+					endGeo.NumOutgoingSegments > 1 &&
+					((Services.SimulationService.LeftHandDrive && endGeo.NumOutgoingLeftSegments > 0) ||
+					(!Services.SimulationService.LeftHandDrive && endGeo.NumOutgoingRightSegments > 0));
```

Comment:
I moved the logic from SegmentGeometry because SegmentGeometry should not provide such complex logic. This is the right place to check whether the turn-on-red feature should be allowed.